File: app/src/test/java/jp/toastkid/yobidashi/libs/UrlsTest.java
package jp.toastkid.yobidashi.libs;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static org.junit.Assert.*;

/**
 * {@link Urls}' test case.
 *
 * @author toastkidjp
 */
@RunWith(RobolectricTestRunner.class)
public class UrlsTest {

    /**
     * {@link Urls#isInvalidUrl(String)}' test.
     * @throws Exception
     */
    @Test
    public void test_isInvalidUrl() throws Exception {
        assertTrue(Urls.Companion.isInvalidUrl(null));
        assertTrue(Urls.Companion.isInvalidUrl(""));
        assertTrue(Urls.Companion.isInvalidUrl("ftp://tomato"));
        assertFalse(Urls.Companion.isInvalidUrl("http://www.yahoo.com"));
        assertFalse(Urls.Companion.isInvalidUrl("https://www.yahoo.com"));
    }

    /**
     * {@link Urls#isValidUrl(String)}' test.
     * @throws Exception
     */
    @Test
    public void test_isValidUrl() throws Exception {
        assertFalse(Urls.Companion.isValidUrl(null));
        assertFalse(Urls.Companion.isValidUrl(""));
        assertFalse(Urls.Companion.isValidUrl("ftp://tomato"));
        assertTrue(Urls.Companion.isValidUrl("http://www.yahoo.com"));
        assertTrue(Urls.Companion.isValidUrl("https://www.yahoo.com"));
    }

}

File: lib/src/test/java/jp/toastkid/lib/UrlsTest.kt
package jp.toastkid.lib

import jp.toastkid.lib.Urls
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner

/**
 * [Urls]' test case.
 *
 * @author toastkidjp
 */
@RunWith(RobolectricTestRunner::class)
class UrlsTest {

    /**
     * [Urls.isInvalidUrl]' test.
     * @throws Exception
     */
    @Test
    fun test_isInvalidUrl() {
        assertTrue(Urls.isInvalidUrl(""))
        assertTrue(Urls.isInvalidUrl("ftp://tomato"))
        assertFalse(Urls.isInvalidUrl("http://www.yahoo.com"))
        assertFalse(Urls.isInvalidUrl("https://www.yahoo.com"))
    }

    /**
     * [Urls.isValidUrl]' test.
     * @throws Exception
     */
    @Test
    fun test_isValidUrl() {
        assertFalse(Urls.isValidUrl(""))
        assertFalse(Urls.isValidUrl("ftp://tomato"))
        assertTrue(Urls.isValidUrl("http://www.yahoo.com"))
        assertTrue(Urls.isValidUrl("https://www.yahoo.com"))
    }

}